### PR TITLE
Add a script for adding multiple testcases in terminal

### PR DESCRIPTION
**Pull request by PJeBeK**
This change is [Reviewable](https://reviewable.io/reviews/cms-dev/cms/647)

### CODE REVIEW

File: cmscontrib/AddTests.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    parser.add_argument("-o", "--overwrite", action="store_true",
+                        help="if tests can overwrite existing tests")
+    parser.add_argument("-c", "--contest_name", action="store",
+                        help="name of contest which tests will be attached to")
+    args = parser.parse_args()
+
+    with SessionGen() as session:
+        dataset = session.query(Dataset)\
+            .join(Dataset.task)\
+            .filter(Task.name == args.task_name).first()
```
Comment (wil93) [inline cmscontrib/AddTests.py]: Taking the `.first()` here just takes a random dataset among this task's datasets, doesn't it?

**Comment by stefano-maggiolo** [comment]
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[setup.py, line 132 \[r1\]](https://reviewable.io:443/reviews/cms-dev/cms/647#-KR7TbCDBEzCOa0FgvkL:-KR7TbCDBEzCOa0FgvkM:b-rfywkf) ([raw file](https://github.com/cms-dev/cms/blob/73125c97917fa3b8427225bb90695107fff962d5/setup.py#L132)):*
> ```Python
>             "cmsAddSubmission=cmscontrib.AddSubmission:main",
>             "cmsAddTeam=cmscontrib.AddTeam:main",
>             "cmsAddTests=cmscontrib.AddTests:main",
> ```

Tests are not testcases, please rename to "AddTestcases"

---

*[cmscontrib/AddTests.py, line 20 \[r1\]](https://reviewable.io:443/reviews/cms-dev/cms/647#-KR7TmAw0-giUQ7jJIcj:-KR7TmAw0-giUQ7jJIck:bcvjpvg) ([raw file](https://github.com/cms-dev/cms/blob/73125c97917fa3b8427225bb90695107fff962d5/cmscontrib/AddTests.py#L20)):*
> ```Python
> # along with this program.  If not, see <http://www.gnu.org/licenses/>.
> 
> """This script add multiple testcases.
> ```

Typo ("adds"), also can you expand a bit (like "...adds multiple testcases from the filesystem to an existing dataset").

---

*[cmscontrib/AddTests.py, line 37 \[r1\]](https://reviewable.io:443/reviews/cms-dev/cms/647#-KR7TxqqB2EBSUw23MUh:-KR7Txqrts7Wg4eVCqov:bejr8bb) ([raw file](https://github.com/cms-dev/cms/blob/73125c97917fa3b8427225bb90695107fff962d5/cmscontrib/AddTests.py#L37)):*
> ```Python
> from cms.db import SessionGen, Dataset, Testcase, Task, Contest
> from cms.db.filecacher import FileCacher
> ```

Two white lines also here :)

---

*[cmscontrib/AddTests.py, line 65 \[r1\]](https://reviewable.io:443/reviews/cms-dev/cms/647#-KQ5bY0ot6ogDExtHkEx-r1-65:-KR7U2V3Nt9jca3OwaoG:bvfsr8o) ([raw file](https://github.com/cms-dev/cms/blob/73125c97917fa3b8427225bb90695107fff962d5/cmscontrib/AddTests.py#L65)):*
<details><summary><i>Previously, wil93 (William Di Luigi) wrote…</i></summary>
> Taking the `.first()` here just takes a random dataset among this task's datasets, doesn't it?
</details>
Yes, you need either to ask for which dataset, or prepopulate with `task.active_dataset_id`.

---

*[cmscontrib/AddTests.py, line 75 \[r1\]](https://reviewable.io:443/reviews/cms-dev/cms/647#-KR7UTFnX05fJZQiw5ZF:-KR7UTFo-Wt3GFdrLbyN:b-j4k800) ([raw file](https://github.com/cms-dev/cms/blob/73125c97917fa3b8427225bb90695107fff962d5/cmscontrib/AddTests.py#L75)):*
> ```Python
>             if task.contest != contest:
>                 print("%s is not in %s" % (args.task_name, args.contest_name))
>                 return False
> ```

Why is this useful?

---

*[cmscontrib/AddTests.py, line 88 \[r1\]](https://reviewable.io:443/reviews/cms-dev/cms/647#-KR7UzqQq48ZNSj4Pc3o:-KR7UzqQq48ZNSj4Pc3p:bpfc0uy) ([raw file](https://github.com/cms-dev/cms/blob/73125c97917fa3b8427225bb90695107fff962d5/cmscontrib/AddTests.py#L88)):*
> ```Python
>         task_name = task.name
> 
>         with zipfile.ZipFile(archive, "r") as archive_zfp:
> ```

I understand you are copying from AWS, so a couple of things:
- it would be better to avoid this duplicate code, maybe put it somewhere in cmscommon?
- we have now cmscommon.archive, which can handle different archive types, if you want to try to use that instead of zipfile (it's fine if you don't).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cms-dev/cms/647)*

**Comment by akmohtashami** [comment]
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[cmscontrib/AddTests.py, line 75 \[r1\]](https://reviewable.io:443/reviews/cms-dev/cms/647#-KR7UTFnX05fJZQiw5ZF:-KR7ZXfZ_j5EuIu737Ar:b-9dnyf2) ([raw file](https://github.com/cms-dev/cms/blob/73125c97917fa3b8427225bb90695107fff962d5/cmscontrib/AddTests.py#L75)):*
<details><summary><i>Previously, stefano-maggiolo (Stefano Maggiolo) wrote…</i></summary>
> Why is this useful?
</details>
You can specify contest id as a fail-safe, so that you won't suddenly add the tests to another task.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cms-dev/cms/647)*

**Comment by stefano-maggiolo** [comment]
Merged!

**Pull request closed without merging**